While event (#3360)
modelica/ModelicaSpecification: chapters/statements.tex

\end{lstlisting}
The rules for \lstinline!for!-statements are the same as for \lstinline!for!-expressions in \cref{explicit-iteration-ranges-of-for-equations} -- except that the \lstinline!expression! of a \lstinline!for!-statement is not restricted to a parameter-expression.

> If the \lstinline!for!-statement contains expressions that will generate events the range should be an evaluable expression.

Reviewer: Stronger:
```suggestion
If the \lstinline!for!-statement contains expressions that will generate events, the range shall be an evaluable expression.
```

> If the \lstinline!for!-statement contains expressions that will generate events, the range shall be an evaluable expression.

Reviewer: Use standard formulation _event-generating expression_, and deal with possibility for multiple expressions in the range:
```suggestion
If the \lstinline!for!-statement contains event-generating expressions, any expression in \lstinline[language=grammar]!for-index! shall be evaluable.
```

\begin{nonnormative}

> The same event-generating expression in different iterations of the for-loop are seen as different crossing functions, and the restriction ensures that their number is known during translation time.

Reviewer: I'm not sure we need this non-normative paragraph, but if we do:
```suggestion
In general, the same event-generating expression requires distinct crossing functions for different iterations of the \lstinline!for!-loop, and the restriction ensures that the number of crossing functions is known during translation time.
```

> \begin{nonnormative}
> In general, the same event-generating expression requires distinct crossing functions for different iterations of the \lstinline!for!-loop, and the restriction ensures that the number of crossing functions is known during translation time.
> \end{nonnormative}

Reviewer: Clarify paragraph breaks:
```suggestion

\begin{nonnormative}
In general, the same event-generating expression requires distinct crossing functions for different iterations of the \lstinline!for!-loop, and the restriction ensures that the number of crossing functions is known during translation time.
\end{nonnormative}

```

\item
  If the \lstinline!expression! of the \lstinline!while!-statement is true, the entire body of the \lstinline!while!-statement is executed (except if a \lstinline!break!-statement, see \cref{break-statement}, or a \lstinline!return!-statement, see \cref{return-statements}, is executed), and then execution proceeds at step~\ref{while-execution-top}.
\end{enumerate}

> Neither the \lstinline!expression! nor any expression in the body shall generate events.

Reviewer: Paragraph break:
```suggestion

Neither the \lstinline!expression! nor any expression in the body shall generate events.
```

In addition to this one, it would be nice to also get a paragraph break before this line:
```
The \lstinline!while!-statement corresponds…
```
(It's funny that we compare with _programing languages_, as if Modelica wasn't a programming language, and as if every programming language has a while loop construct.)

Reviewer: Also use standard formulation:
```suggestion

Event-generating expressions are neither allowed in the \lstinline!expression! nor in the loop body statements.
```

Author: > In addition to this one, it would be nice to also get a paragraph break before this line:
> 
> ```
> The \lstinline!while!-statement corresponds…
> ```
> 
> (It's funny that we compare with _programing languages_, as if Modelica wasn't a programming language, and as if every programming language has a while loop construct.)

I added line break and changed to _other_ programming languages.

> A deprecated feature is that there is an implicit \lstinline!noEvent! for such \lstinline!while!-statement.

Reviewer: It's the expressions that are wrapped in `noEvent`:
```suggestion
A deprecated feature is that all expressions in a \lstinline!while!-statement are implicitly inside \lstinline!noEvent!.
```

\item
  If the \lstinline!expression! of the \lstinline!while!-statement is true, the entire body of the \lstinline!while!-statement is executed (except if a \lstinline!break!-statement, see \cref{break-statement}, or a \lstinline!return!-statement, see \cref{return-statements}, is executed), and then execution proceeds at step~\ref{while-execution-top}.
\end{enumerate}
Neither the \lstinline!expression! nor any expression in the body shall generate events.

> In order to ensure that there are no events it is recommended to use explicit \lstinline!noEvent! in \lstinline!while!-statements in models.

Reviewer: This sentence is just strange, and it following statement about deprecated behavior is sufficient:
```suggestion
```
